### PR TITLE
Add Native Mobile First Page

### PR DESCRIPTION
**Pull request by BenBJSS**

### CODE REVIEW

File: app/section/sections/mobile-first/ui.test.js [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      .reply(200, {});
+  }
+};
+
+const pageSetup = async (t, withMarketingData = false) => {
+  mocks(withMarketingData);
+  await t.navigateTo('http://localhost:1234/S100000-001/section/mobile-first');
+};
+
+fixture.only('Show Plug-ins Or Extensions page')
```
Comment (sam-roberts-bjss) [inline app/section/sections/mobile-first/ui.test.js]: Need to remove the '.only'

Comment (sam-roberts-bjss) [reply app/section/sections/mobile-first/ui.test.js]: Also the fixture name needs changing

File: app/section/sections/mobile-first/ui.test.js [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  const additionalAdvice = Selector('[data-test-id="section-additional-advice"]');
+
+  await t
+    .expect(additionalAdvice.innerText).eql(expectedAdditionalAdvice);
+});
+
+test('should render the mobile first design question', async (t) => {
+  await pageSetup(t);
+
+  const pluginsRequiredQuestion = Selector('[data-test-id="question-mobile-first-design"]');
```
Comment (kat1906) [inline app/section/sections/mobile-first/ui.test.js]: rename variable

File: app/section/sections/mobile-first/ui.test.js [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  const submitButton = Selector('[data-test-id="section-submit-button"]');
+
+  await t
+    .expect(submitButton.find('button').count).eql(1);
+});
+
+test('should populate the questions with existing data', async (t) => {
+  pageSetup(t, true);
+
+  const pluginsRequiredQuestion = Selector('[data-test-id="question-mobile-first-design"]');
```
Comment (kat1906) [inline app/section/sections/mobile-first/ui.test.js]: same here - rename variable

File: app/section/sections/mobile-first/ui.test.js [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+  const yesRadiobutton = pluginsRequiredQuestion.find('.nhsuk-radios__item:nth-child(1)');
+  const noRadiobutton = pluginsRequiredQuestion.find('.nhsuk-radios__item:nth-child(2)');
+
+  await t
+    .expect(yesRadiobutton.find('input:checked').exists).ok()
+    .expect(noRadiobutton.find('input:checked').exists).notOk();
+});
+
+test('should show error summary and validation for plugins required question indicating it is mandatory', async (t) => {
```
Comment (kat1906) [inline app/section/sections/mobile-first/ui.test.js]: rename test

File: app/section/sections/mobile-first/ui.test.js [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+  nock('http://localhost:8080')
+    .put('/api/v1/Solutions/S100000-001/sections/mobile-first')
+    .reply(400, {
+      required: ['mobile-first-design'],
+    });
+
+  const errorSummary = Selector('[data-test-id="error-summary"]');
+  const errorSummaryList = Selector('.nhsuk-error-summary__list');
+  const pluginsRequiredQuestion = Selector('[data-test-id="question-mobile-first-design"]');
```
Comment (kat1906) [inline app/section/sections/mobile-first/ui.test.js]: same here - rename variable

File: app/section/sections/mobile-first/ui.test.js [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+    .expect(additionalAdvice.innerText).eql(expectedAdditionalAdvice);
+});
+
+test('should render the mobile first design question', async (t) => {
+  await pageSetup(t);
+
+  const pluginsRequiredQuestion = Selector('[data-test-id="question-mobile-first-design"]');
+
+  await t
+    .expect(pluginsRequiredQuestion.find('.nhsuk-fieldset__legend').innerText).eql('Was this Solution design with a mobile first approach?')
```
Comment (sam-roberts-bjss) [inline app/section/sections/mobile-first/ui.test.js]: Careful with variable names. The variable 'pluginsRequiredQuestion' could do with a better name

File: app/section/sections/mobile-first/ui.test.js (hunk 3 above)
Comment (sam-roberts-bjss) [inline app/section/sections/mobile-first/ui.test.js]: Variable name check

File: app/section/sections/mobile-first/ui.test.js (hunk 4 above)
Comment (sam-roberts-bjss) [inline app/section/sections/mobile-first/ui.test.js]: Check test name

File: app/section/sections/mobile-first/ui.test.js (hunk 5 above)
Comment (sam-roberts-bjss) [inline app/section/sections/mobile-first/ui.test.js]: Variable name check

**Pull request merged**